src/sqlancer/common/log/SQLLoggableFactory.java

```diff
@@ -24,7 +24,11 @@ protected Loggable createLoggable(String input, String suffix) {
 
     @Override
     public SQLQueryAdapter getQueryForStateToReproduce(String queryString) {
-        return new SQLQueryAdapter(queryString);
+        if (queryString.contains("CREATE TABLE")) {
```

Comment:
IMO changes could take place in the default constructor of `SQLQueryAdapter`, not here. That class takes care of those stuffs, i.e. `sqlancer.common.query.SQLQueryAdapter#checkQueryString`.

Reply:
Agree. Such changes are more reasonable.

Reply:
Does the following change confuse other users?
```Java
    public SQLQueryAdapter(String query) {
        this(query, new ExpectedErrors(), query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN"));
    }
```

Reply:
I'd use something like:
```
    public SQLQueryAdapter(String query, ExpectedErrors expectedErrors) {
        this(query, expectedErrors, guessAffectSchemaFromQuery(query));
    }

    private static boolean guessAffectSchemaFromQuery(String query) {
        return query.contains("CREATE TABLE") && !query.startsWith("EXPLAIN");
    }

    private void checkQueryString() {
        if (guessAffectSchemaFromQuery(query) != couldAffectSchema) {
            throw new AssertionError("CREATE TABLE statements should set couldAffectSchema to true");
        }
    }
```

Reply:
Agree. Thanks for help.
I change the committed code.
PTAL